src/main/java/UtilsService.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    public List<String> splitTextIntoChunks(String text, int chunkSize) {
+        List<String> chunks = new ArrayList<>();
+        int length = text.length();
+        for (int i = 0; i < length; i += chunkSize) {
+            chunks.add(text.substring(i, Math.min(length, i + chunkSize)));
+        }
+        return chunks;
+    }
+
+    public Map<String, Object> retryParsingWithSmallerChunks(String responseText) {
+        log.info("Retrying parsing with smaller chunks...");
+
+        // Split the response text into chunks of 1000 characters
+        int chunkSize = 1000;  // You can adjust this size based on the typical response size
+        List<String> chunks = splitTextIntoChunks(responseText, chunkSize);
+        List<Map<String, Object>> fileSuggestions = new ArrayList<>();
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        // Process each chunk individually
+        for (String chunk : chunks) {
+            try {
+                // Try parsing each chunk individually
+                Map<String, Object> chunkSuggestions = mapper.readValue(chunk, new TypeReference<Map<String, Object>>() {});
+                List<Map<String, Object>> files = (List<Map<String, Object>>) chunkSuggestions.get("files");
+
+                // If the chunk has valid files, add them to fileSuggestions
+                if (files != null && !files.isEmpty()) {
+                    for (Map<String, Object> fileObj : files) {
+                        fileSuggestions.add(fileObj);
+                    }
+                }
+            } catch (JsonParseException e) {
+                log.warn("Skipping invalid chunk due to JSON parsing error: " + e.getMessage());
+            } catch (Exception e) {
+                log.error("Error processing chunk: " + e.getMessage());
+            }
+        }
+
+        // Return the combined suggestions from all valid chunks
+        if (!fileSuggestions.isEmpty()) {
+            return Map.of("fileSuggestions", fileSuggestions);
+        } else {
+            return Map.of("status", "error", "message", "Unable to parse the GPT response, even in smaller chunks.");
+        }
+    }
+
+    public String extractCodeFromDiff(String diff) {
+        StringBuilder codeBuilder = new StringBuilder();
+        String[] diffLines = diff.split("\n");
+
+        // Skip lines that start with diff markers or non-code elements
+        for (String line : diffLines) {
+            // Skip file mode, diff metadata, and line indicators
+            if (!line.startsWith("diff") && !line.startsWith("index") && !line.startsWith("---") && !line.startsWith("+++")
+                    && !line.startsWith("@@") && !line.contains("new file mode")) {
+
+                // Strip out leading '+' or '-' which indicate additions/deletions in the diff
+                if (line.startsWith("+") || line.startsWith("-")) {
+                    codeBuilder.append(line.substring(1)).append("\n");
```

Comment:
This operation assumes that the line is not empty after stripping the leading character. It could lead to an IndexOutOfBoundsException. (Severity: High)